core-plugins/src/main/java/io/cdap/plugin/transform/ProjectionTransform.java

```diff
@@ -138,6 +146,25 @@ public void initialize(TransformContext context) throws Exception {
     collector.getOrThrowException();
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) {
+    init(context.getInputSchema(), context.getFailureCollector());
+
+    List<FieldOperation> output = new ArrayList<>();
+    Collection<String> fieldsToKeep = this.fieldsToKeep;
+    if (fieldsToKeep.isEmpty()) {
+      fieldsToKeep = TransformLineageRecorderUtils.getFields(context.getInputSchema());
+    }
+    output.addAll(TransformLineageRecorderUtils.generateOneToOnes(new ArrayList<>(fieldsToKeep),
+      "identity", TransformLineageRecorderUtils.IDENTITY_TRANSFORM_DESCRIPTION));
+    output.addAll(TransformLineageRecorderUtils.generateDrops(new ArrayList<>(fieldsToDrop)));
```

Comment:
for drops, if the fieldsToKeep is not empty, any field that is not in the keep list is dropped. See https://github.com/cdapio/hydrator-plugins/blob/087c35db5174f754d1a5fba6abc209ee645f864e/core-plugins/src/main/java/io/cdap/plugin/transform/ProjectionTransform.java#L175-L176